installer/environment.py

```
@@ -62,6 +99,8 @@ def generate_ssh_key(self):
                 )
                 if result.returncode != 0:
                     raise Exception("Failed to generate SSH key")
+            except FileNotFoundError as e:
+                raise Exception(f"ssh-keygen not found: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)